x/assets/types/general.go

```diff
@@ -68,6 +69,18 @@ type DeltaOperatorSingleAsset OperatorAssetInfo
 
 type CreateQueryContext func(height int64, prove bool) (sdk.Context, error)
 
+// StakerBalance is a struct to describe the balance of a staker for a specific asset
+// balance = withdrawable + delegated + pendingUndelegated
```

Comment:
`pendingUndelegated` represents the expected amount. The actual undelegated amount might be less if some pending undelegations are slashed. It's better to add this code comment to prevent issues when using this function.

Reply:
add some comments to clarify this concept